Gruntfile.js

```diff
@@ -259,9 +259,9 @@ module.exports = function( grunt ) {
 					banner: "/*! jQuery v<%= pkg.version %> | " +
 						"(c) JS Foundation and other contributors | jquery.org/license */",
 					compress: {
+						"pure_getters": true,
```

Comment:
Hmm, what does it exactly do? The Uglify README mentions:

> UglifyJS will assume that object property access (e.g. `foo.bar` or `foo["bar"]`) doesn't have any side effects

but technically we can't guarantee that as `jQuery.event.addProp` creates a getter utilizing a provided hook.

Reply:
My understanding is from going to something like:

```js
foo.bar++;
if (foo.bar) {
  console.log('test');
}
```
to something like:

```js
++foo.bar && console.log("test");
```
In the first case, If you have a getter on "bar" it will be executed twice, in the second case – once i.e. we should be fine

Reply:
I see, that meaning of the flag makes sense. On the other hand, I imagine something like the following:
```js
var bar,
    flag = foo.a;
if (flag) {
    bar = flag + 1;
}
```
might be minified to (I'm skipping other optimizations here):
```js
var bar;
if (foo.a) {
    bar = foo.a + 1;
}
```
which would invoke a getter twice which - in turn - may lead to a performance penalty as the getter is invoked twice.

Reply:
Well, we don't have getters anyway

Reply:
We do, our event object is implemented via getters now.

Reply:
Those event getters are super fast and should be pure functions. It is also normal and expected that they will be called many times.

The potential of calling those getters multiple times for the benefit of size reduction sounds good to me.

Reply:
OK, I've checked what's the difference between the output with `pure_getters` on and off and it was huge, 57 bytes gzipped. I prettyfied the code then and compared the two outputs and I saw that one of the differences is `pure_getters` cuts out this:
https://github.com/jquery/jquery/blob/3.3.1/external/sizzle/dist/sizzle.js#L193-L195
```js
	// Support: Android<4.0
	// Detect silently failing push.apply
	arr[ preferredDoc.childNodes.length ].nodeType;
```
and this: https://github.com/jquery/jquery/blob/3.3.1/external/sizzle/dist/sizzle.js#L1521-L1523
```js
			// Accessing this property makes selected-by-default
			// options in Safari work properly
			if ( elem.parentNode ) {
				elem.parentNode.selectedIndex;
			}
```
etc.

In other words, in our code getters are sometimes invoked for side effects only so we cannot enable this flag.

Reply:
Nice find @mgol, 'pure_getters' option is removed